app/components/channel_link/channel_link.test.js

```diff
@@ -7,26 +7,33 @@ import {Text} from 'react-native';
 
 import ChannelLink from './channel_link';
 
+jest.mock('react-intl');
+
 describe('ChannelLink', () => {
+    const formatMessage = jest.fn();
     const channelsByName = {
-        firstChannel: {id: 'channel_id_1', name: 'firstChannel', display_name: 'First Channel'},
-        secondChannel: {id: 'channel_id_2', name: 'secondChannel', display_name: 'Second Channel'},
+        firstChannel: {id: 'channel_id_1', name: 'firstChannel', display_name: 'First Channel', team_id: 'current_team_id'},
+        secondChannel: {id: 'channel_id_2', name: 'secondChannel', display_name: 'Second Channel', team_id: 'current_team_id'},
     };
     const baseProps = {
         channelName: 'firstChannel',
+        currentTeamId: 'current_team_id',
+        currentUserId: 'current_user_id',
         linkStyle: {color: '#2389d7'},
         onChannelLinkPress: jest.fn(),
         textStyle: {color: '#3d3c40', fontSize: 15, lineHeight: 20},
         channelsByName,
         actions: {
             handleSelectChannel: jest.fn(),
+            joinChannel: jest.fn(),
             setChannelDisplayName: jest.fn(),
         },
     };
 
     test('should match snapshot', () => {
         const wrapper = shallow(
-            <ChannelLink {...baseProps}/>
+            <ChannelLink {...baseProps}/>,
+            {context: {intl: {formatMessage}}},
```

Comment:
Don't we have a `shallowWithIntl` utility to do this for us?

Reply:
Not sure why but it's not working for me.  I'm getting below error:
``Warning: Failed context type: The context `intl.formatDate` is marked as required in `ChannelLink`, but its value is `undefined`.``

I'll try to investigate why it's not working as it should be.